extern/boostd-data/svc/svc_test.go

```diff
@@ -340,14 +346,129 @@ func compareIndices(subject, subjectDb index.Index) (bool, error) {
+	defer cancel()
+
+	t.Run("level db", func(t *testing.T) {
+		bdsvc, err := NewLevelDB("")
+		require.NoError(t, err)
+		testCleanup(ctx, t, bdsvc, 8042)
+	})
+	t.Run("couchbase", func(t *testing.T) {
+		// Enable it back once SB tests are working on CI
+		t.Skip(t)
+		removeContainer := setupCouchbase(t)
+		defer removeContainer()
+		bdsvc := NewCouchbase(testCouchSettings)
+		testCleanup(ctx, t, bdsvc, 8043)
+	})
+}
+
+func testCleanup(ctx context.Context, t *testing.T, bdsvc *Service, port int) {
+	err := bdsvc.Start(ctx, port)
+	require.NoError(t, err)
+
+	cl := client.NewStore()
+	err = cl.Dial(context.Background(), "http://localhost:8042")
+	require.NoError(t, err)
+	defer cl.Close(ctx)
+
+	sampleidx := "fixtures/baga6ea4seaqnfhocd544oidrgsss2ahoaomvxuaqxfmlsizljtzsuivjl5hamka.full.idx"
+
+	pieceCid, err := cid.Parse("baga6ea4seaqnfhocd544oidrgsss2ahoaomvxuaqxfmlsizljtzsuivjl5hamka")
+	require.NoError(t, err)
+
+	subject, err := loadIndex(sampleidx)
+	require.NoError(t, err)
+
+	records, err := getRecords(subject)
+	require.NoError(t, err)
+
+	randomuuid, err := uuid.Parse("4d8f5ce6-dbfd-40dc-8b03-29308e97357b")
+	require.NoError(t, err)
+
+	err = cl.AddIndex(ctx, pieceCid, records)
+	require.NoError(t, err)
+
+	di := model.DealInfo{
+		DealUuid:    randomuuid,
+		SectorID:    abi.SectorNumber(1),
+		PieceOffset: 1,
+		PieceLength: 2,
+		CarLength:   3,
+	}
+
+	// Add a deal for the piece
+	err = cl.AddDealForPiece(ctx, pieceCid, di)
+	require.NoError(t, err)
+
+	// There should only be one deal
+	dis, err := cl.GetPieceDeals(ctx, pieceCid)
+	require.NoError(t, err)
+	require.Len(t, dis, 1)
+	require.Equal(t, di, dis[0])
+
+	b, err := hex.DecodeString("1220ff63d7689e2d9567d1a90a7a68425f430137142e1fbc28fe4780b9ee8a5ef842")
+	require.NoError(t, err)
+
+	mhash, err := multihash.Cast(b)
+	require.NoError(t, err)
+
+	offset, err := cl.GetOffsetSize(ctx, pieceCid, mhash)
+	require.NoError(t, err)
+	require.EqualValues(t, 3039040395, offset.Offset)
+	require.EqualValues(t, 0, offset.Size)
+
+	pcids, err := cl.PiecesContainingMultihash(ctx, mhash)
+	require.NoError(t, err)
+	require.Len(t, pcids, 1)
+	require.Equal(t, pieceCid, pcids[0])
+
+	indexed, err := cl.IsIndexed(ctx, pieceCid)
+	require.NoError(t, err)
+	require.True(t, indexed)
+
+	recs, err := cl.GetRecords(ctx, pieceCid)
+	require.NoError(t, err)
+	require.Equal(t, len(records), len(recs))
+
+	loadedSubject, err := cl.GetIndex(ctx, pieceCid)
+	require.NoError(t, err)
+
+	ok, err := compareIndices(subject, loadedSubject)
+	require.NoError(t, err)
+	require.True(t, ok)
+
+	err = cl.RemoveDealForPiece(ctx, pieceCid, di.DealUuid)
+	require.NoError(t, err)
+
+	_, err = cl.GetPieceDeals(ctx, pieceCid)
+	require.ErrorContains(t, err, "not found")
+
+	_, err = cl.GetOffsetSize(ctx, pieceCid, mhash)
+	require.ErrorContains(t, err, "not found")
+
+	_, err = cl.GetRecords(ctx, pieceCid)
+	require.ErrorContains(t, err, "not found")
```

Comment:
Let's also add checks that PiecesContainingMultihash and IsIndexed work as expected